Mixin: Transparentize

machycek opened the pull request:
Features that are available with `transparentize` function 🎉

- [x] **non transparent** color (rgb, hex, hsl) to transparent color
- [x] **transparent** color (rgba, hsla) to "more" transparent color

nikgraf commented on src/mixins/transparentize.js:
```diff
@@ -0,0 +1,57 @@
+  )
+)
+
+function transparentize(color: string, percentage : number) {
```
Can you make sure it accepts a range from 0 to 1 instead of 0 to 100.

Why?
I believe it's easier to comprehend for people that if the current value is 0.8 that by passing 0.2 it would end up with 0.6. Having an 0.8 alpha and passing in 20 to reduce it to 0.6 feels strange to me. Thats also the reason why saturation, lightness is using 0 to 1 and we should stay consistent across the lib.

Can you also remove the space after percentage.

machycek replied:
Got it. I guess you don't want to use percentage then? At the moment everything is based on percentage eg. transparentize('rgba(255, 255, 255, .5)', 20) => rgba(255, 255, 255, .4)

machycek replied:
Should I change it from **"percentage"** to **"amount"** then?

nikgraf replied:
@machycek yes please 👍

nikgraf commented on src/mixins/transparentize.js:
```diff
@@ -0,0 +1,57 @@
+    const hslValues = getColorValuesInBrackets(color)
+    const newTransparency = getNewTransparency(1, percentage)
+
+    return `hsla(${hslValues}, ${newTransparency})`
```
Right now most of the other functions accept only rgba or hex strings as this is the output from our rgb, rgba, hsl & hsla functions. So even hsla outputs rgba as described here https://github.com/styled-components/polished/issues/44

> - We want to create the smallest possible output. For colors without opacity we use hex e.g. `#FF553A`. If possible only 3 characters `#FFF` instead of `#FFFFFF`. In case of opacity we generate `rgba(255,0,0,0.3)`.

Can you make sure transparentize also exports those? The internal helper contain a couple of helpers that does this conversion.

It might make sense to support `hsl`, `hsla`, `hwb` and `lab` ([all valid CSS](https://drafts.csswg.org/css-color/#the-hwb-notation)) in the future, but I'm not sure. There is a tradeoff. Being open to more inputs means increase code size.

machycek replied:
Sounds good, going to change that 👍🏼

nikgraf commented on src/mixins/transparentize.js:
```diff
@@ -0,0 +1,57 @@
+    ),
+  )
+)
+
```
Can you add a comment like these? https://github.com/styled-components/polished/blob/master/src/helpers/hsla.js#L7-L32

We use it to generate the docs.

machycek commented:
Glad to hear that, thanks Nik!

machycek commented:
Ok, so I have done following

- moved transparentize from mixins to helpers
- added comments for the **docs**
- changed the range from 0 - 100 to **0 - 1**
- renamed percentage to amount
- changed logic from percentage to **amount**
- changed order of arguments (amount is **first**, color is last) 
- changed output of hsl / hsla to **rgba**

Looking forward for feedback 👍🏼
